pkg/objstore/s3/s3.go

```diff
@@ -44,6 +46,11 @@ const (
 
 	// SSES3 is the name of the SSE-S3 method for objstore encryption.
 	SSES3 = "SSE-S3"
+
+	// SSEConfigKey is the context key to override SSE config. This feature is used by downstream
+	// projects (eg. Cortex) to inject custom SSE config on a per-request basis. Future work or
+	// refactoring can introduce breaking changes as far as the functionality is preserved.
```

Comment:
Let's add a comment maybe, a NOTE that this is context value only because it's a very specific S3 option. If SSE is available wider we should probably add a variadic option as (https://github.com/thanos-io/thanos/issues/3784#issuecomment-778330460)  WDYT?